src/v/storage/segment.h

```diff
@@ -220,6 +225,11 @@ class segment {
 
     absl::btree_map<size_t, model::offset> _inflight;
 
+    // Timestamp from server time of first data written to this segment,
+    // field is set when a raft_data batch is appended.
+    // Used to implement segment.ms
```

Comment:
nit: maybe explicitly note that this _isn't_ the first timestamp of a data batch in the log, and is only set if the segment was appended to while this segment was active. I.e. a closed segment following a restart wouldn't have this value set, and that's okay because we only intend on using this in the context of an active segment. Perhaps it'd be better to add as a comment to the `first_write_ts()` method so callers are aware.

Otherwise I can imagine this eventually being accidentally used improperly to make decisions based on the age of the data.

Reply:
If you don't mind, I used what you wrote with minor modifications for the comment of first_write_ts()